spec/factories/session.rb

```diff
@@ -1,12 +1,15 @@
-
 FactoryBot.define do
   factory :session do
     sequence :title do |n|
-      " Session #{n}"
+      "Session #{n}"
     end
     description { 'whatever' }
     participant
     association :event
-    association :room
+
+    # ensure room is associated with the same event as the session
+    after(:build) do |session|
+      session.room = session.event.rooms.first || create(:room, event: session.event)
+    end
```

Comment:
This fixes a bug that was exposed by the addition of counter caches - whereby a room getting created would inadvertently create an additional event, thus `Event.current_event` wasn't what the test was expecting.